core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/validation/ValidateRequest.java

```diff
@@ -159,34 +155,29 @@ private ValidationResult doValidate(BranchContext context, Writer index) throws
 						final Collection<ComponentIdentifier> ruleWhiteListEntries = whiteListedEntries.removeAll(ruleId);
 						final String toolingId = rules.stream().filter(rule -> ruleId.equals(rule.getId())).findFirst().get().getToolingId();
 						for (ValidationIssueDetails issueDetails : ruleIssues.issueDetails) {
+							final ValidationIssue validationIssue;
 							ComponentIdentifier componentIdentifier = issueDetails.affectedComponentId;
 							
 							if (!existingIsssuesByComponentIdentifier.containsKey(componentIdentifier)) {
-								final ValidationIssue validationIssue = new ValidationIssue(
+								validationIssue = new ValidationIssue(
 										UUID.randomUUID().toString(),
 										ruleId,
-										branchPath,
-										componentIdentifier,
+										ComponentURI.of(codeSystemURI, componentIdentifier.getTerminologyComponentId(), componentIdentifier.getComponentId()),
```

Comment:
Let's have a factory method on `ComponentURI` that receives a `ComponentIdentifier`, so it's easier to convert one from the other. Add it to the other PR #707